### DISCUSSION TITLE
Playbook Admin Login - Not working

### DISCUSSION THREAD

**Discussion by SasquatchSecurity** [Security-Onion-Solutions/securityonion#6127] [Unsupported Versions] [answered]
When accessing https://securityonion/playbook/login and attempting to login as "admin" using the password from `sudo salt-call pillar.get secrets`, I get an "Internal Error" page. Below is the output from playbook.log during the attempted login process. 

Running 2.3.80. All services are up. Installed via ISO 2.3.80 image as a standalone node virtualized via Proxmox. 4 cores / 16GB ram. Everything is up to date using soup. I've even done so-playbook-reset to no avail. I should note that I had an issue where one play got stuck in "Disabled" which is why I wanted to login with admin. Since performing the reset, all plays are left in Draft but some still alert on the SOC page despite being in "Draft". 

### `sudo salt-call state.highstate` output:
```bash
local:
    Data failed to compile:
----------
    The function "state.highstate" is running as PID 29780 and was started at 2021, Nov 06 18:17:53.294847 with jid 20211106181753294847
```
### `playbook.log` output:
```bash
App 109 output: I, [2021-11-06T18:16:09.527884 #109]  INFO -- : Started GET "/playbook/login" for 172.17.0.1 at 2021-11-06 18:16:09 +0000
App 109 output: I, [2021-11-06T18:16:09.528581 #109]  INFO -- : Processing by AccountController#login as HTML
App 109 output: I, [2021-11-06T18:16:09.530651 #109]  INFO -- :   Current user: anonymous
App 109 output: I, [2021-11-06T18:16:09.531779 #109]  INFO -- :   Rendering account/login.html.erb within layouts/base
App 109 output: I, [2021-11-06T18:16:09.532482 #109]  INFO -- :   Rendered account/login.html.erb within layouts/base (0.6ms)
App 109 output: I, [2021-11-06T18:16:09.540148 #109]  INFO -- :   Rendered plugins/redmine_playbook/app/views/playbook/_footer.html.erb (0.0ms)
App 109 output: I, [2021-11-06T18:16:09.540314 #109]  INFO -- : Completed 200 OK in 12ms (Views: 7.1ms | ActiveRecord: 2.5ms)
App 109 output: I, [2021-11-06T18:17:02.212683 #109]  INFO -- : Started POST "/playbook/login" for 172.17.0.1 at 2021-11-06 18:17:02 +0000
App 109 output: I, [2021-11-06T18:17:02.213220 #109]  INFO -- : Processing by AccountController#login as HTML
App 109 output: I, [2021-11-06T18:17:02.213264 #109]  INFO -- :   Parameters: {"utf8"=>"✓", "authenticity_token"=>"REDACTED", "back_url"=>"/playbook/projects/detection-playbooks/issues", "username"=>"admin", "password"=>"[FILTERED]", "login"=>"Login"}
App 109 output: I, [2021-11-06T18:17:02.215777 #109]  INFO -- :   Current user: anonymous
App 109 output: I, [2021-11-06T18:17:02.220236 #109]  INFO -- : Completed 500 Internal Server Error in 7ms (ActiveRecord: 3.9ms)
App 109 output: F, [2021-11-06T18:17:02.220718 #109] FATAL -- :
App 109 output: F, [2021-11-06T18:17:02.220749 #109] FATAL -- : ActiveModel::MissingAttributeError (missing attribute: twofa_scheme):
App 109 output: F, [2021-11-06T18:17:02.220762 #109] FATAL -- :
App 109 output: F, [2021-11-06T18:17:02.220775 #109] FATAL -- : app/models/user.rb:406:in `twofa_active?'
App 109 output: app/controllers/account_controller.rb:321:in `password_authentication'
App 109 output: app/controllers/account_controller.rb:307:in `authenticate_user'
App 109 output: app/controllers/account_controller.rb:41:in `login'
App 109 output: lib/redmine/sudo_mode.rb:61:in `sudo_mode'
```

**Comment by defensivedepth** [comment, marked as answer]
Hello there!

A couple things:
Run the following on the Manager: `sudo so-playbook-restart --force` and then try to login with the admin creds.

As for the alerts still being generated - Run the following on the Manager - it will delete all the old Play rules that may still be active.
`rm -f /opt/so/rules/elastalert/playbook/`

**Reply by SasquatchSecurity** [reply]
@defensivedepth thanks for the tips. Interesting that even after a reboot, the admin login wouldn't work, but a forced restart fixed it. I am in, so that's great news. 

As for the plays that are still triggering, I cleared the yaml files from that directory (I left the .template files). Time will tell if it still triggers, but I suspect not. Is there a better way to ensure Playbook stays in sync with what's in that directory? Obviously performing a reset isn't typical, but I would've thought it'd clear that directory of all old plays and/or did a check against what was present vs. what was Active.

**Reply by defensivedepth** [reply]
`so-playbook-sync` runs every 5 min and keeps those files in sync with `Active` Plays. If you are still seeing issues, I would suggest looking at the logs for this.
